src/NServiceBus.AcceptanceTests/Sagas/When_using_ReplyToOriginator.cs

```diff
@@ -76,7 +102,8 @@ class MyReplyToOriginatorHandler : IHandleMessages<MyReplyToOriginator>
 
                 public Task Handle(MyReplyToOriginator message, IMessageHandlerContext context)
                 {
-                    TestContext.Intent = (MessageIntentEnum) Enum.Parse(typeof(MessageIntentEnum), context.MessageHeaders[Headers.MessageIntent]);
+                    TestContext.Intent = (MessageIntentEnum)Enum.Parse(typeof(MessageIntentEnum), context.MessageHeaders[Headers.MessageIntent]);
+                    TestContext.ReceivedCorrelationId = context.MessageHeaders[Headers.CorrelationId];
                     TestContext.Done = true;
```

Comment:
we can remove the Done condition and check whether `ReceivedCorrelationId` has a value